src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,28 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | null = null;
+    @Output() readonly buyedProduct: EventEmitter<Product | null> = new EventEmitter();
+
+    getProductImageUrl(index: number): string | null {
+        return this.product?.images && this.product?.images?.length > index
+            ? this.product?.images[index]?.url
+            : null;
```

Comment:
Я бы рекомендовал поступить так:
```suggestion
        if (!this.product?.images) {
             return null;
        }
        
        const {images} = this.product;
        
        return images.length > index ? images[index].url : null;
```

Reply:
исправила

Reply:
Тут немного все таки поясню, почему рекомендовал сделать именно так: код декомпозирован на 2 этапа, что повышает читаемость + за счет TS сужается типизация, что позволяет нам работать именно с `this.product: Product` после `if`, а не с `this.product: Product | undefined`